File: lab2.java

class MyThread extends Thread {
    public static void main(String[] args) {
        Thread t = new Thread(() -> System.out.println("Hello from a thread!"));
 
        t.start();
    }
}


class lab2 {
    public static void main(String[] args) {
        for (int i = 0; i < 10; i++) {
            Thread t = new Thread(() -> System.out.println("Hello from a thread!"));
            t.start();
            try {
                t.join();
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        }
        System.out.println("Hello from main!");
    }
}

File: lab2.kt
internal object MyThread : Thread() {
    fun main(args: Array<String?>?) {
        val t = Thread { System.out.println("Hello from a thread!") }
        t.start()
    }
}

internal object lab2 {
    fun main(args: Array<String?>?) {
        for (i in 0..9) {
            val t = Thread { System.out.println("Hello from a thread!") }
            t.start()
            try {
                t.join()
            } catch (e: InterruptedException) {
                e.printStackTrace()
            }
        }
        System.out.println("Hello from main!")
    }
}